### PR TITLE
feat: add commitlint

### LINKED ISSUE #625: Install commitlint through pre-commit hook
See https://github.com/OWASP/wrongsecrets/discussions/624 : let's implement https://github.com/alessandrojcm/commitlint-pre-commit-hook :)

### PR DESCRIPTION
**Pull request by bendehaan**
Closes #625 

What kind of changes does this PR include?

-   [x] Fixes or refactors
-   [ ] A new challenge
-   [x] Additional documentation
-   [x] Something else

Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [ ] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

**Review by commjoen — CHANGES_REQUESTED** [review]
Would you be so kind to update the readme.md and contributing.md on the process of installing precommit first please?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,12 +70,14 @@ Pull requests should be as small/atomic as possible. Large, wide-sweeping change
...
-10. Your PR will be reviewed and comments may be given. In order to process a comment, simply make modifications to the same branch as before and push them to your repository. GitHub will automatically detect these changes and add them to your existing PR.
-11. When starting on a new PR in the future, make sure to always keep your local repo up to date:
+8. Install [pre-commit](https://pre-commit.com/#install) the dependencies for our pre-commit configuration to make sure your code complies with standards used in the project. This requires terraform, [terraform-docs](https://github.com/terraform-docs/terraform-docs#installation), [tflint](https://github.com/terraform-linters/tflint#installation), and [commitlint](https://commitlint.js.org/#/guides-local-setup). For commitlint, you can use `npm install` in the root folder of this project.
```
Comment (commjoen) [inline CONTRIBUTING.md:75]:
```suggestion
8. Install [pre-commit](https://pre-commit.com/#install) the dependencies for our pre-commit configuration to make sure your code complies with standards used in the project. This requires terraform, [terraform-docs](https://github.com/terraform-docs/terraform-docs#installation), [tflint](https://github.com/terraform-linters/tflint#installation), and [commitlint](https://commitlint.js.org/#/guides-local-setup). For commitlint, you need to have [node](https://nodejs.org/en/download/) installed, after which you can use `npm install` in the root folder of this project.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -70,12 +70,14 @@ Pull requests should be as small/atomic as possible. Large, wide-sweeping change
...
+9. Install the pre-commit hook using `pre-commit install --hook-type commit-msg`. We recommend to run `pre-commit run -a` every so often if you're working on a bigger change.
```
Comment (commjoen) [inline CONTRIBUTING.md:76]: Can we maybe install all hooks at the same time ;) as a plan b?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -70,12 +70,14 @@ Pull requests should be as small/atomic as possible. Large, wide-sweeping change
...
+10. After your modifications are done, push them to your forked repository. This can be done by executing the command `git add MYFILE` for every file you have modified, followed by `git commit -m 'your commit message here'` to commit the modifications and `git push` to push your modifications to GitHub.
+11. Create a Pull Request (PR) by going to your fork, <https://github.com/Your_Github_Handle/wrongsecrets> and click on the "New Pull Request" button. The target branch should typically be the Master branch. When submitting a PR, be sure to follow the checklist that is provided in the PR template. The checklist itself will be filled out by the reviewer.
+12. Your PR will be reviewed and comments may be given. In order to process a comment, simply make modifications to the same branch as before and push them to your repository. GitHub will automatically detect these changes and add them to your existing PR.
+13. When starting on a new PR in the future, make sure to always keep your local repo up to date:
```
Comment (commjoen) [inline CONTRIBUTING.md:80]: maybe in the end : what if something went wrong? whjat if you have to redo your commit messages?

File: CONTRIBUTING.md (hunk 2 above)
Comment (bendehaan) [reply CONTRIBUTING.md:76]: All commit hooks are installed at the same time, but they are specifically executed on commit msg end with this command. Double pre-commit installs will look iffy / cause multiple runs.

File: CONTRIBUTING.md (hunk 3 above)
Comment (bendehaan) [reply CONTRIBUTING.md]: Addressed with extra line

**Pull request merged**